src/software/ai/navigator/path_planner/path_planner_test.cpp

```diff
@@ -0,0 +1,255 @@
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(5 * M_PI / 3), std::sin(5 * M_PI / 3)}),
+                      false),
+              },
+          .should_return_path = true},
+
+         {.name           = "Circle of robots surrounding friendly robot at distance 0.2",
+          .start          = Point(0, 0),
+          .dest           = Point(4, 0),
+          .navigable_area = Rectangle({-5, -5}, {5, 5}),
+          .obstacles =
+              {
+                  Obstacle::createRobotObstacle(TestUtil::createRobotAtPos({0.2, 0}),
+                                                false),
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(M_PI / 3) * 0.2, std::sin(M_PI / 3) * 0.2}),
+                      false),
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(2 * M_PI / 3) * 0.2, std::sin(2 * M_PI / 3) * 0.2}),
+                      false),
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(3 * M_PI / 3) * 0.2, std::sin(3 * M_PI / 3) * 0.2}),
+                      false),
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(4 * M_PI / 3) * 0.2, std::sin(4 * M_PI / 3) * 0.2}),
+                      false),
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(5 * M_PI / 3) * 0.2, std::sin(5 * M_PI / 3) * 0.2}),
+                      false),
+              },
+          .should_return_path = false}};
+
+template <typename PlannerT>
+std::pair<std::string, PathPlannerConstructor> name_and_constructor()
+{
+    return std::pair<std::string, PathPlannerConstructor>(
+        typeid(PlannerT).name(), []() { return std::make_unique<PlannerT>(); });
+}
+
+std::vector<std::pair<std::string, PathPlannerConstructor>>
+    path_planner_names_and_constructors = {
+        // add path planner constructors here
+        name_and_constructor<ThetaStarPathPlanner>(),
+        // uncomment this if you want some tests to fail
+        //        name_and_constructor<StraightLinePathPlanner>()
+};
+
+
+void validatePath(const Path &path, const Point &start, const Point &dest,
+                  const Rectangle &navigable_area, const std::vector<Obstacle> &obstacles)
+{
+    // check for zero length path
+    if (path.valueAt(0.f) == path.valueAt(1.f))
+    {
+        throw std::string("zero length path!");
+    }
+
+    // compute an s value interval that roughly corresponds to length
+    // PATH_CHECK_INTERVAL_M assuming that the path is locally linear
+    double s_per_meter_ratio   = 0.05 / (path.valueAt(0.05) - path.valueAt(0.0)).length();
```

Comment:
Should 0.05 be replaced with `PATH_CHECK_INTERVAL_M`?

Reply:
0.05 is a fixed, small s value only used to approximate s/meter, and isn't relevant to `PATH_CHECK_INTERVAL_M`, which is the interval in metres at which to evaluate the spline.